ci/tests/puppeteer/scenarios/oauth2-login-jwt-at/script.js

```diff
@@ -63,6 +63,25 @@ async function executeFlow(browser, redirectUri, clientId, accessTokenSecret) {
             throw error;
         });
 
+    const parts = accessToken.split('.');
+    let badAccessToken;
+    if (parts.length === 3) {
+        badAccessToken = parts[0] + '.Z' + parts[1] + '.' + parts[2];
+    } else {
+        badAccessToken = parts[0] + '.' + parts[1] + '.Z' + parts[2] + '.' + parts[3] + '.' + parts[4];
+    }
+    const badParams = new URLSearchParams();
+    badParams.append('access_token', badAccessToken);
```

Comment:
Could I ask, why are you doing this to create a bad access token? Isn't something like `kwdhfkusdyfksdbfkjdsgfhjsdgfhkjhseriuyewiurhskjfbcdkjfbdsjk` enough?

Reply:
Or if you need it to be a jwt, why not create a proper JWT? What exists here is very black magic.

Reply:
In fact, to produce the `Unable to parse JWT` exception, I needed to change the content (not the header or signature) of the JWT.

Reply:
I'd very much recommend that you add a comment here and explain what happens here then. 5 months from now, it's virtually impossible to figure out what's being tested here. Even better, you could refactor this into a "createAccessTokenThatDoesXYZ" so comments would be unnecessary.

Reply:
I don't want a proper JWT here, I want a malformed JWT. This is why I create it manually.

Reply:
I have added an explanation in the Puppeteer test.